train.py

```diff
@@ -0,0 +1,302 @@
+                os.path.join(data_dir, "ours", "negative-keyword-examples"),
+                "-type",
+                "f",
+                "-name",
+                "*.wav",
+            ]
+        )
+        .decode("utf-8")
+        .split("\n")
+    )
+    away_negative_keyword_examples = (
+        subprocess.check_output(
+            [
+                "find",
+                os.path.join(data_dir, "ours", "away-negative-keyword-examples"),
+                "-type",
+                "f",
+                "-name",
+                "*.wav",
+            ]
+        )
+        .decode("utf-8")
+        .split("\n")
+    )
+    # Call gsutil ls command to get all files in the bucket, this is done because call them
+    # with find command is too slow
+
+    google_dataset = (
+        subprocess.check_output(
+            [
+                "gsutil",
+                "ls",
+                "gs://autonomy-vision/audios/external-datasets/google_dataset/**.wav",
+            ]
+        )
+        .decode("utf-8")
+        .split("\n")
+    )
+    mswc = (
+        subprocess.check_output(
+            [
+                "gsutil",
+                "ls",
+                "gs://autonomy-vision/audios/external-datasets/mswc/**.wav",
+            ]
+        )
+        .decode("utf-8")
+        .split("\n")
+    )
+
+    negative_files = list(
+        set(
+            negative_keyword_examples
+            + away_negative_keyword_examples
+            + google_dataset
+            + mswc
+        )
+    )
+    negative_files.remove("")
+    negative_files = [f for f in negative_files if ".wav" in f]
+    print(f"Founded {len(negative_files)} negative files")
+
+    # Convert the obtained URIs to local path
+    external_dataset_path = data_dir / "external"
+    external_dataset_path_str = str(external_dataset_path)
+
+    negative_files = [
+        f.replace(
+            "gs://autonomy-vision/audios/external-datasets",
+            external_dataset_path_str,
+        )
+        for f in negative_files
+    ]
+
+    # Convert files to Path
+    positive_files = [Path(f) for f in positive_files]
+    negative_files = [Path(f) for f in negative_files]
+
+    if toy:
+        print(" --- Using toy dataset ---")
+        positive_files = positive_files[:200]
+        negative_files = negative_files[:2000]
+
+    model_dir = file_path.parent.parent.parent / "models" / "kwd" / "embedding_model"
+    # base_model = tf.keras.models.load_model(model_dir)
+    # embedding = tf.keras.models.Model(
+    #     name="embedding_model",
+    #     inputs=base_model.inputs,
+    #     outputs=base_model.get_layer(name="dense_2").output,
+    # )
+    # embedding.trainable = False
+
+    # Prepare data
+    negative_files_str = [str(path) for path in negative_files]
+
+    # Split the list of files into train, validation, and test sets
+    train_negatives, test_negatives = train_test_split(
+        negative_files_str, test_size=0.1, random_state=42
+    )
+
```

Comment:
:( add a TODO that we should split this better to avoid data leakage

Reply:
I added it a few lines after this

Reply:
Done